### PR TITLE
Remove the advanced functionality of the 404 page (Blockpage)

### PR DESCRIPTION
**Pull request by PromoFaux**
**By submitting this pull request, I confirm the following:** 

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md), as well as this entire template.
- [x] I have made only one major change in my proposed changes.
- [ ] I have commented my proposed changes within the code.
- [x] I have tested my proposed changes, and have included unit tests where possible.
- [x] I am willing to help maintain this change if there are issues with it later.
- [x] I give this submission freely and claim no ownership.
- [x] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [x] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

---
**What does this PR aim to accomplish?:**

There are several motivations behind this, namely:

- IP blocking is no longer the default blocking mode for new installs of Pi-hole. So most people will never see it.
- Even if a user is configured to IP blocking mode - the block page will only be displayed on a `HTTP` request only. 
  - With the above point in mind, as the internet is mainly [`HTTPS` these days](https://transparencyreport.google.com/https/overview?hl=en) - most people will never see it
- I've been meaning to do this for ages
- From a maintenance point of view - it's just not worth keeping that functionality of the 404 handler

That all said, the 404-handling aspect of the page _is_ retained, e.g auto redirect from `http://pi.hole` to `http://pi.hole/admin`.

If a user now has blocking mode set to IP, and given the example of `blockedpage.com` being on their block list, this is what they will now see when visiting `http://blockedpage.com`

image

Users can still create their own 404 handler (to include "blockpage" fuctionality if they so wish, though per the notes above - why would you?) by creating a file name `custom.php` in `/var/www/html/pihole` and designing it to their tastes. (if they really wanted, they could just copy paste the code from the old `index.php` into `custom.php`, but they would be responsible for making it work if anything broke 😉 )

It was a nice page when it was first invented, but I think the time has finally come to say goodbye to an old friend, and put it out of it's misery.

### CODE REVIEW

**Comment by dschaper** [comment]
I like the ability to include a custom page, that's nice for backwards compatibility. I've labeled it to 6.0, but that's preliminary and open for debate. I consider it to be a big enough "breaking" change to need to be on the next version but other viewpoints are welcome.

**Comment by PromoFaux** [comment]
`custom.php` support has been there forever, I don't know how many people actually use it. 

I also realised that the "did you mean to go to the admin page" thing can be overridden by creating a page called `landing.php`

There is a path to backward compatibility, so I wouldn't say it's a major breaking change (in that if someone really wants to use it they can just use the `custom.php` path. But certainly not the next point release. So `6.0`, or `5.3`.. whichever comes first!

**Comment by DL6ER** [comment]
I'm very much in favor of this change. Why are some things lines
``` php
// Set mobile friendly viewport
$viewPort = '<meta name="viewport" content="width=device-width, initial-scale=1">';
```
still there? Are they needed?

**Comment by PromoFaux** [comment]
> still there? Are they needed?

Possibly missed it. I will give it a second pass :)

File: advanced/index.php [hunk 1]
```diff
@@ -70,10 +46,14 @@ function setHeader($type = "x") {
     <html lang='en'>
         <head>
             <meta charset='utf-8'>
-            $viewPort
+            <meta name='viewport' content='width=device-width, initial-scale=1'>
             <title>● $serverName</title>
-            <link rel='stylesheet' href='pihole/blockingpage.css'>
             <link rel='shortcut icon' href='admin/img/favicons/favicon.ico' type='image/x-icon'>
```
Comment (MichaIng) [inline advanced/index.php:51]: Just throwing in the question why asset paths are relative, starting with this PR: #3263
This breaks asset access when accessing a non-existent resource inside a sub directory as `https://my.domain.org/subdir/admin/img/favicons/favicon.ico` in this particular case does not exist. Same with the logo below. Inconsistently the admin page link is still absolut?

**Comment by PromoFaux** [comment]
rebased on latest `development`

File: advanced/index.php [hunk 2]
```diff
@@ -34,21 +25,6 @@
...
-// Get extension of current URL
-$currentUrlExt = pathinfo($_SERVER["REQUEST_URI"], PATHINFO_EXTENSION);
-
-// Set mobile friendly viewport
-$viewPort = '<meta name="viewport" content="width=device-width, initial-scale=1">';
-
-// Set response header
-function setHeader($type = "x") {
-    header("X-Pi-hole: A black hole for Internet advertisements.");
-    if (isset($type) && $type === "js") header("Content-Type: application/javascript");
```
Comment (yubiuser) [inline advanced/index.php:49]: The debug script checks for this x-header. If you remove it here, also remove the test from the debug script

Comment (MichaIng) [reply advanced/index.php:49]: But it only does this for the admin panel, while here it is for the blocking/404 page, isn't it?

Comment (yubiuser) [reply advanced/index.php:49]: No, for both
https://github.com/pi-hole/pi-hole/blob/dad6247cb0c90fe8cc1ad541e0ccf8f02e1dd2be/advanced/Scripts/piholeDebug.sh#L771

File: advanced/index.php (hunk 1 above)
Comment (DL6ER) [reply advanced/index.php:52]: @PromoFaux

**Comment by dschaper** [comment]
MC needs to be resolved.

File: automated install/basic-install.sh [hunk 3]
```diff
@@ -1781,18 +1781,18 @@ install_dependent_packages() {
...
-    install -D -m 644 ${PI_HOLE_LOCAL_REPO}/advanced/{index,blockingpage}.* ${PI_HOLE_BLOCKPAGE_DIR}/
+    # Install the directory
+    install -d -m 0755 ${PI_HOLE_404_DIR}
+    # and the 404 handler
+    install -D -m 644 ${PI_HOLE_LOCAL_REPO}/advanced/index.php ${PI_HOLE_404_DIR}/
 
     # Remove superseded file
-    if [[ -e "${PI_HOLE_BLOCKPAGE_DIR}/index.js" ]]; then
-        rm "${PI_HOLE_BLOCKPAGE_DIR}/index.js"
+    if [[ -e "${PI_HOLE_404_DIR}/index.js" ]]; then
```
Comment (yubiuser) [inline automated install/basic-install.sh:1794]: I would remove this part entirely, it was added 5 years ago by https://github.com/pi-hole/pi-hole/pull/1724 and released with v3.2. Do you really think someone is upgrading straight from v3.1?

Comment (yubiuser) [reply automated install/basic-install.sh:1794]: Or you use it to remove the now upersedded `blockingpage.css`

File: advanced/index.php (hunk 1 above)
Comment (yubiuser) [reply advanced/index.php]: https://github.com/pi-hole/pi-hole/pull/3910/commits/adaff1793cb5b64b28ac4e71a2f70a572edb77b7

Comment (yubiuser) [reply advanced/index.php]: https://github.com/pi-hole/pi-hole/pull/3910/commits/adaff1793cb5b64b28ac4e71a2f70a572edb77b7

File: advanced/index.php [hunk 4]
```diff
@@ -51,16 +51,24 @@
...
-              #splashpage { background: #222; color: rgba(255, 255, 255, 0.7); text-align: center; }
-              #splashpage img { margin: 5px; width: 256px; }
-              #splashpage b { color: inherit; }
+                html, body { height: 100% }
+                body { font: 13pt "Source Sans Pro", "Helvetica Neue", Helvetica, Arial, sans-serif; }
+                body { margin: 0; }
+                p { margin: 0; }
+                a { color: #3c8dbc; text-decoration: none; }
+                a:hover { color: #72afda; text-decoration: underline; }
+                #splashpage { background: #222; color: rgba(255, 255, 255, 0.7); text-align: center; display: flex; align-items: center; justify-content: center; }
```
Comment (rdwebdesign) [inline advanced/index.php:60]: Here, `body` and `#splashpage` are the same element.

This is an inline CSS and it'll only be used within this page, so we can do something like this:

```suggestion
                html, body { height: 100% }
                body { margin: 0; font: 13pt "Source Sans Pro", "Helvetica Neue", Helvetica, Arial, sans-serif; }
                body { background: #222; color: rgba(255, 255, 255, 0.7); text-align: center; }
                p { margin: 0; }
                a { color: #3c8dbc; text-decoration: none; }
                a:hover { color: #72afda; text-decoration: underline; }
                #splashpage { display: flex; align-items: center; justify-content: center; }
```

**Review by rdwebdesign — DISMISSED** [review]

**Comment by PromoFaux** [comment]
I will try and give this another look over tomorrow with the changes added - just a reminder that it will probably need a bit of squashing!

**Comment by yubiuser** [comment]
Doc changes needed here: https://docs.pi-hole.net/ftldns/blockingmode/

**Review by yubiuser — APPROVED** [review]

**Review by yubiuser — DISMISSED** [review]

**Review by yubiuser — APPROVED** [review]

**Review by yubiuser — APPROVED** [review]

**Comment by yubiuser** [comment]
Documentation PR: https://github.com/pi-hole/docs/pull/765

**Pull request merged**